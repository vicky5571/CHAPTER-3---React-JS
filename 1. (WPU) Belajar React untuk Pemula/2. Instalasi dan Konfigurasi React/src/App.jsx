/* eslint-disable react/prop-types */
import { useState } from "react";

function Header({ author }) {
  return <h1>Belajar React Bareng {author ? author : "WPU"}</h1>;
}

function App() {
  const students = ["Vicky", "Galih", "Pamungkas"];
  const [likes, setLikes] = useState(0);

  function handleClick() {
    setLikes(likes + 1);
  }
  return (
    <>
      <Header author="Vicky" />
      <ul>
        {students.map((student) => (
          <li key={student}>{student}</li>
        ))}
      </ul>
      <button onClick={handleClick}>Like ({likes})</button>
    </>
  );
}

export default App;
