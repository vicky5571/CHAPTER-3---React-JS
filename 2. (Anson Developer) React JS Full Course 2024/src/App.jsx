/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
import { useEffect } from "react";
import { PostContainer } from "./components/PostContainer";
import { UserContext } from "./components/utils/contexts/UserContext";
import { PostContentButton } from "./components/utils/contexts/UserContext";
import { Outlet } from "react-router-dom";

export default function App() {
  const { user, loading, error } = useFetchUser(2);
  console.log(user, loading, error);

  useEffect(() => {
    if (!loading && !error && user) setUserData(user);
  }, [loading, error, user]);
  return (
    <div>
      <>
        <UserContext.Provider value={{ ...userData, setUserData }}>
          <div>{loading ? "Loading..." : <PostContainer />}</div>
        </UserContext.Provider>
        <PostContentButton />
        <Outlet />
      </>
    </div>
  );
}
