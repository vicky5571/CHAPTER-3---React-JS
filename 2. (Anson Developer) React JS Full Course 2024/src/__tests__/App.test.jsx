import { expect, it, describe } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "../App";

describe("App", () => {
  it("should render users", () => {
    const { container } = render(<App usersData={[]} />);
    expect(container).toMatchSnapshot();
  });

  describe("Edit Button is Clicked", () => {
    it("should render save button", async () => {
      render(
        <App
          usersData={[
            {
              id: 1,
              username: "vicky5571",
              email: "vicky@gmail.com",
            },
          ]}
        />
      );
      const editButton = screen.getByRole("button", { name: "Edit" });
      await userEvent.click(editButton);
      const saveButton = screen.getByRole("button", { name: "Save" });
      expect(saveButton).toBeInTheDocument();
    });

    it("should display username & email input fields", async () => {
      render(
        <App
          usersData={[
            {
              id: 1,
              username: "vicky5571",
              email: "vicky@gmail.com",
            },
          ]}
        />
      );
      const editButton = screen.getByRole("button", { name: "Edit" });
      await userEvent.click(editButton);
      expect(screen.getByLabelText("Username:")).toBeInTheDocument();
      expect(screen.getByLabelText("Email:")).toBeInTheDocument();
      //   expect(screen.getByRole("textbox", { name: "username" })).toBeInTheDocument();
      //   expect(screen.getByRole("textbox", { name: "email" })).toBeInTheDocument();
    });
  });
});
