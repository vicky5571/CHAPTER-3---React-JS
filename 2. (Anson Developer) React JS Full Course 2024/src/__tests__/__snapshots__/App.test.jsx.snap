// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render users 1`] = `<div />`;
